### ISSUE TITLE
[Crunchyroll] Some videos (only seen on One Piece so far) are not downloading correctly.

### ISSUE THREAD

**Issue by almostalive** [rg3/youtube-dl#8418]
I'm not sure if this is an issue that can be fixed in youtube-dl or if its just an error on crunchyroll's side. Quite a few of One Piece episodes starting at 607 will download the flv stream instead of the mp4 file that is normally downloaded. I noticed this while watching the show through mpv that It wasn't able to fill the cache and then tried to download the episode with youtube-dl. I saw that the extension was flv instead of mp4 and that it started downloading very slow and only got up to a little over 1MiB/s, normally it caps out my connection. I tried a lot of random episodes after that one and it seems to be random on the episodes that work right or not. I didn't check any episodes before it since all the ones that I have watched this way worked.

So not sure if this is limited to One Piece or it is possibly happening on other shows.
Here is a gist with verbose and dump-pages. Hopefully something useful is in it.
https://gist.github.com/almostalive/184b571350a0a197667c
@dstftw since it will most likely be you that looks at this, this is happening on a US connection not sure if it affects other regions.

and a random episode link that has the same problem http://www.crunchyroll.com/one-piece/episode-703-a-rocky-road-law-and-corazons-journey-of-life-681809

**Comment by remitamine** [comment]
the http links for this video saved in a different hosting(v17.ak.crunchyroll.com)
like this: http://v17.ak.crunchyroll.com/ve1916845/video.mp4?__gda__=1454590413_1dca253534cd26f45a431b46db1683ff
and this: http://v17.ak.crunchyroll.com/ve1471915/video.mp4?__gda__=1454590072_a26b6908df8fa5bb6bf5668d939c80f6

**Comment by almostalive** [comment]
Interesting, Thanks for that info. So I'm guessing this could be fixed on youtube-dl's side. All the affected videos look to be using edgefcs.net so a workaround like in commit: a01da8bbf83dfd4f87e3fdd105b9f7c850e76cad could work. Sadly my lack of knowledge of python and how ytdl works, I wasn't able to get anything to work.

**Comment by dstftw** [comment]
I don't see any problem here. rtmp from edgefcs downloads and plays fine. fplive trick was a necessity to workaround rtmp streams that rtmpdump is unable to cope with. Moreover I'm not aware of such workaround for edgefcs.

**Comment by almostalive** [comment]
I decided to go ahead and download the rest of the episodes instead of trying to watch them streaming through mpv and dealing with it not being able to cache the bad flv files correctly. Went with the mp4 files first to get them done quickly. This is the command I used:
youtube-dl http://www.crunchyroll.com/one-piece -f mp4 --playlist-start=815 --write-sub --sub-lang 'enUS' --embed-subs --merge-output-format 'mp4' --download-archive onepiece.txt -i
This is starting at episode 607. YTDL give us a total number of: [crunchyroll:playlist] playlist One Piece: Collected 939 video ids (downloading 125 of them)
Since I no longer have a Premium account I can't download the current newest video, so lets make this 124. 64* out of 124 downloaded the mp4 format from v.lvlt.crcdn.net server just fine. Leaving 60 videos that I've got to download very slowly in flv format from edgefcs.net server. It just seems strange that these 60 episodes are using a different server than the rest and are scattered throughout. But they do download and play just fine so I guess I won't push the issue anymore. Thanks for looking into it though.

* Episode 723 ( http://www.crunchyroll.com/one-piece/episode-723-a-collision-of-haki-luffy-vs-doflamingo-681849 ) downloads both the video and subtitle files just fine but ffmpeg errrors out on trying to merge them. Ran the command that ytdl gives ffmpeg by it self and it just Segmentation fault out. But that is more likely an issue for ffmpeg. I just thought it was weird and worth mentioning. All the other files merged correctly execpt of course the flv files since they can't.
